packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -69,15 +69,32 @@ export const useCheckoutPage = props => {
 
     const {
         data: checkoutData,
-        called: checkoutCalled,
-        loading: checkoutLoading
+        networkStatus: checkoutQueryNetworkStatus
     } = useQuery(getCheckoutDetailsQuery, {
+        /**
+         * Skip fetching checkout details if the `cartId`
+         * is a falsy value.
+         */
         skip: !cartId,
+        notifyOnNetworkStatusChange: true,
         variables: {
             cartId
         }
     });
 
+    /**
+     * For more info about network statues check this out
+     *
+     * https://www.apollographql.com/docs/react/data/queries/#inspecting-loading-states
+     */
+    const isLoading = useMemo(() => {
+        const checkoutQueryInFlight = checkoutQueryNetworkStatus
+            ? checkoutQueryNetworkStatus < 7
+            : true;
```

Comment:
Since [network status](https://www.apollographql.com/docs/react/api/react-apollo/#datanetworkstatus) is 1-based, here's an alternate approach that reads a little easier in my opinion:

```js
const checkoutQueryInFlight = !checkoutQueryNetworkStatus || checkoutQueryNetworkStatus < 7;
```

Reply:
Nice approach Andy, but I would like to stick to the ternary operator approach. In my opinion, in this case, it is easy to read than the `||` approach.